Tasks/AzureFunctionV2/src/request-builder.ts

```diff
@@ -0,0 +1,95 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+import { TaskInputs } from './types';
+import { AxiosRequestConfig } from 'axios';
+import { AzureRMEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azure-arm-endpoint';
+import { AuthType } from './constants';
+import { AzureEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azureModels';
+
+export class RequestBuilder {
+  constructor(private inputs: TaskInputs) {}
+
+  public async buildRequest(): Promise<AxiosRequestConfig> {
+    return this.inputs.authType === AuthType.Key ? this.buildKeyAuthRequest() : await this.buildArmAuthRequest();
+  }
+
+  private buildKeyAuthRequest(): AxiosRequestConfig {
+    tl.debug(`Building key authentication request for function: ${this.inputs.function}`);
+    let url = `${this.inputs.function}?code=${this.inputs.key}`;
+
+    // Add query parameters if present
+    if (this.inputs.queryParameters) {
+      url += this.inputs.queryParameters;
```

Comment:
The description of query param input in task.json says that input should not start with &. However, if you just concatenate it here it will not produce valid url. E.g.:

```
https://function.azure.com?code=xxxxxxqueryparam1=value&queryparam2=value
```

I would not rely on the fact that users wont put & or & as first character in query parameter input. Lets trim it or maybe even parse it and use only valid parts.